CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using AngleSharp;
+using CryptoWorld.News.Data;
+using CryptoWorld.News.Data.Models;
+using CryptоWorld.News.Core.Interfaces;
+using CryptоWorld.News.Core.ViewModels.Home_Page;
+using System.Globalization;
+using System.Text;
+
+namespace CryptоWorld.News.Core.Services.News
+{
+    public class NewsService : INewsService
+    {    
+        private readonly AngleSharp.IConfiguration config;
+        private readonly IBrowsingContext context;
+        private List<PageNewsModel> homeNews;
+        private List<string> urls;
+        private readonly ApplicationDbContext dbContext;
+        public NewsService( ApplicationDbContext _dbContext)
+        {
+            config = Configuration.Default.WithDefaultLoader();
+            context = BrowsingContext.New(config);
+            homeNews = new List<PageNewsModel>();
+            urls = new List<string>();
+            dbContext = _dbContext;
+        }
+
+        public async Task<List<PageNewsModel>> HomePageNews(int pagesCount = 7)
```

Comment:
Could you please refactor this method by adding more methods? It looks like it does a lot of things. For example, you can separate the implementation into different methods - GetNewsUrls, GetPageNewsModel, GetArticles etc...

Reply:
The method is splitted